### PR TITLE
Ember lvl 1

### PR DESCRIPTION
**Pull request by Rxbsxn**
[Rendered](https://github.com/Selleo/DevPath/blob/fa84054140edd32ccba701ea3e2ad9e8d08d5f4a/frontend_developer/ember/01_level_1.md)

### CODE REVIEW

**Comment by PoslinskiNet** [comment]
@Rxbsxn I would also add some exercise, as we have for [NodeJS Path](https://github.com/Selleo/selleo_best_practices/blob/master/service_delivery_paths/backend_developer/nodejs/dev_path.md) with requirements, so we make sure that someone can connect dots together.

**Comment by Rxbsxn** [comment]
@PoslinskiNet Yea, I've it at back of my head but first I want to create meaningful description to each section.

File: frontend_developer/ember/01_level_1.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+**Ember CLI**
+- Start a new application
+- Addons
+- Commands
+- Application file structure
+- Build
+
+**Templates & Helpers**
+- Understand the implication that handlebars is not javascript
```
Comment (mikoscz) [inline frontend_developer/ember/01_level_1.md:20]: X for doubt

File: frontend_developer/ember/01_level_1.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+ - use beforeModel, model, afterModel, redirect hooks
+
+**Components**
+- Communication with the outside world: Data Down | Actions Up | Services
+- Customize Component:
+-    html: splatattributes
+- handling actions:
+-    component events
+-    actions hash / how to access?
+- Component Properties: element, elementId
```
Comment (mikoscz) [inline frontend_developer/ember/01_level_1.md:51]: Those are no longer component properties

File: frontend_developer/ember/01_level_1.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+ - use nested route and use dynamic segments (router, route)
+ - use beforeModel, model, afterModel, redirect hooks
+
+**Components**
+- Communication with the outside world: Data Down | Actions Up | Services
+- Customize Component:
+-    html: splatattributes
+- handling actions:
+-    component events
+-    actions hash / how to access?
```
Comment (mikoscz) [inline frontend_developer/ember/01_level_1.md:50]: Accessing actions hash directly should be avoided - it is still there for the backward compatibility.

File: frontend_developer/ember/01_level_1.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+ - use application, index, wildcard route
+ - use nested route and use dynamic segments (router, route)
+ - use beforeModel, model, afterModel, redirect hooks
+
+**Components**
+- Communication with the outside world: Data Down | Actions Up | Services
+- Customize Component:
+-    html: splatattributes
+- handling actions:
+-    component events
```
Comment (mikoscz) [inline frontend_developer/ember/01_level_1.md:49]: What does this one mean?

Comment (BobrImperator) [reply frontend_developer/ember/01_level_1.md:49]: I believe it relates to the ancient world of ember, where components had their implicit element and they exposed
`onClick`
`onHover`
methods to handle user input.
This was never encouraged by community though, so I don't think we should teach this either.

Let's explicitly point to using template helpers/modifiers and action decorator instead.

File: frontend_developer/ember/01_level_1.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+-    actions hash / how to access?
+- Component Properties: element, elementId
+- tracked property
+- Component Lifecycle
```
Comment (BobrImperator) [inline frontend_developer/ember/01_level_1.md:53]: There's no such thing anymore, component lifecycle methods should always be discouraged.
We should teach how to use modifiers for initialization **when needed**, people love to abuse lifecycle methods therefore I believe it should be moved to advanced section.

File: frontend_developer/ember/01_level_1.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+- What data management libraries can you use instead of the default one?
+
+### 📝 Katas
+- Present a finished tutorial application (Optional)
+
+---
+
+## 📦 Ember / Ember CLI
+
+Optimalize your work with Ember CLI in your toolbelt
```
Comment (BobrImperator) [inline frontend_developer/ember/01_level_1.md:92]:
```suggestion
Optimize your work with Ember CLI in your toolbelt
```

File: frontend_developer/ember/01_level_1.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+- 📗 [Ember CLI](https://cli.emberjs.com/release/)
+- 📗 [Basic use](https://cli.emberjs.com/release/basic-use/)
+- 📗 [Ember observer](https://emberobserver.com/)
+- 📗 [Addons](https://cli.emberjs.com/release/#whatareaddons)
+- 📗 [Application file structure](https://cli.emberjs.com/release/basic-use/folder-layout/)
+
+### 🎤 Interview
+
+- How can we install Ember CLI and create a new project?
+- What are addons to Ember application? Where can you look for interesting package and how to install it?
```
Comment (BobrImperator) [inline frontend_developer/ember/01_level_1.md:105]: Let's split it
```suggestion
- What are addons to Ember application?
- Where can you look for interesting package and how to install it?
```

File: frontend_developer/ember/01_level_1.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+### 🎓 Learn
+
+- 📗 [Templates introduction](https://guides.emberjs.com/release/components/)
+- 📗 [Lists of elements](https://guides.emberjs.com/release/components/looping-through-lists/)
+- 📗 [Conditionals in templates](https://guides.emberjs.com/release/components/conditional-content/)
+- 📗 [Helpers](https://guides.emberjs.com/release/components/helper-functions/)
+
+### 🎤 Interview
+
+- How can you display properties in the template
```
Comment (BobrImperator) [inline frontend_developer/ember/01_level_1.md:126]: I understand that interviewer can use it as a *trick* question, but we should explicitly ask about
`@arguments` `properties` and `attributes`

File: frontend_developer/ember/01_level_1.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
+# Ember - level I
+
+## Areas
+
+**Ember Basic Concepts I**
+- Application structure
+- Ember Object
```
Comment (BobrImperator) [inline frontend_developer/ember/01_level_1.md:7]: I don't think knowledge about Ember Object itself is required.
We should teach Ember/Glimmer Components instead, Ember Object should be treated as private.

**Comment by PoslinskiNet** [comment]
@Rxbsxn can we push MVP in July? (Ember 1,2,3)?

**Pull request merged**